stdlib/public/Concurrency/TaskPrivate.h

```diff
@@ -561,12 +567,31 @@ struct AsyncTask::PrivateStorage {
 
     this->~PrivateStorage();
   }
+
+  swift::atomic<ActiveTaskStatus> &_status() {
```

Comment:
My C++ foo is weak so I'd appreciate a sanity check, I think reinterpret_cast is what we'd want here to treat the storage as the actual atomic?